File: app/src/main/java/com/kota/Bahamut/Pages/Model/ClassPageBlock.java
package com.kota.Bahamut.Pages.Model;

import com.kota.Bahamut.ListPage.TelnetListPageBlock;
import java.util.Stack;

public class ClassPageBlock extends TelnetListPageBlock {
    private static final Stack<ClassPageBlock> _pool = new Stack<>();
    public int mode = 0;

    private ClassPageBlock() {
    }

    public static void release() {
        synchronized (_pool) {
            _pool.clear();
        }
    }

    public static void recycle(ClassPageBlock block) {
        synchronized (_pool) {
            _pool.push(block);
        }
    }

    public static ClassPageBlock create() {
        ClassPageBlock block = null;
        synchronized (_pool) {
            if (_pool.size() > 0) {
                block = _pool.pop();
            }
        }
        if (block == null) {
            return new ClassPageBlock();
        }
        return block;
    }
}

File: app/src/main/java/com/kota/Bahamut/pages/model/ClassPageBlock.kt
package com.kota.Bahamut.pages.model

import com.kota.Bahamut.listPage.TelnetListPageBlock
import java.util.Stack

class ClassPageBlock private constructor() : TelnetListPageBlock() {
    var mode: Int = 0

    companion object {
        private val _pool = Stack<ClassPageBlock?>()
        @JvmStatic
        fun release() {
            synchronized(_pool) {
                _pool.clear()
            }
        }

        @JvmStatic
        fun recycle(block: ClassPageBlock?) {
            synchronized(_pool) {
                _pool.push(block)
            }
        }

        fun create(): ClassPageBlock {
            var block: ClassPageBlock? = null
            synchronized(_pool) {
                if (_pool.size > 0) {
                    block = _pool.pop()
                }
            }
            if (block == null) {
                return ClassPageBlock()
            }
            return block
        }
    }
}
